src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.bookapp.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookapp.dto.BookDto;
+import mate.academy.bookapp.dto.CreateBookRequestDto;
+import mate.academy.bookapp.mapper.BookMapper;
+import mate.academy.bookapp.model.Book;
+import mate.academy.bookapp.repository.BookRepository;
+import mate.academy.bookapp.service.BookService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can`t find the book with id" + id));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookMapper.toDtoList(bookRepository.findAll());
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);

```

Comment:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Can't find the book with id" + id));
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Where in BookMapper
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`